### DISCUSSION TITLE
Darstellung mehrerer Erzeugeranlagen

### DISCUSSION THREAD

**Discussion by StevieC121176** [solectrus/solectrus#2053] [General]
Ich habe zum erstmal im EVCC Forum von diesem Project gehört und bin begeistert. Ich habe 2 Kostal Wechselrichter mit einer BYD Batterie. Da das Kostal Solar Portal echt mies ist war ich schon lange auf der Suche nach einem anständigen Dashboard. 
Heute kam nun der neue Raps und die Installation war eigentlich recht einfach, obwohl ich von Docker keinerlei Ahnung habe.

Die Daten hole ich mir von EVCC über den MQTT Broker des IOBrokers.

Nun meine Frage. Ich habe 2 Wechselrichter. EVCC gibt auch über MQTT die einzelnen Leistungen der PV Anlagen aus. Ist es möglich in Solectrus die einzelnen Anlagen darzustellen oder nur als Gesamtanlage?

Wäre auch ein Farbschema angelehnt an EVCC möglich? Also gelb für Einspeisung, grün Erzeugung und blau für Netzbezug?
und ein Hell/Dunkel Theme?

Auf alle Fälle erstmal Daumen hoch für dieses großartige Programm.

**Comment by ledermann** [comment]
Immer schön, wenn jemand begeistert ist :)

Bezüglich der zwei Wechselrichter: SOLECTRUS geht derzeitig von **einem** PV-Erzeuger aus. Das hat historische Gründe, da SOLECTRUS ursprünglich für die Anbindung an einen SENEC-Speicher entstanden ist.

Grundsätzliche würde ich daher zunächst empfehlen, die Leistung einzelner Anlagen zusammenzufassen (möglicherweise kann das evcc erledigen) und den Gesamtertrag zum MQTT-Broker zu schicken. In SOLECTRUS würde es dann so aussehen, als wäre es **eine** Anlage.

Unabhängig davon ist ein Feature in Vorbereitung, separate MPP-Strings einzeln darzustellen (siehe solectrus/solectrus#1756). Es wäre denkbar, dies weiter zu verallgemeinern - also dass in der InfluxDB neben der Gesamtleistung auch Einzelleistungen gespeichert werden. Ob das Strings einer einzelnen Anlage oder verschiede Anlagen sind, wäre dann egal. 

Bezüglich der Farben. Das Thema hatten wir schon mal, siehe solectrus/solectrus#1859. Ich bin da etwas skeptisch, aber jeder einzelne Ruf nach einem solchem Feature erhöht die Wahrscheinlichkeit, dass ich das doch mal angehe ;-)

Ansonsten: Wenn Du weitere Erfahrungen mit dem MQTT-Collector und SOLECTRUS machst, schreibe gerne darüber. Entweder hier oder lege bei Problemen ein [Issue](https://github.com/solectrus/mqtt-collector/issues) an. Mich würde sehr interessieren, wie sich diese neue Komponente im praktischen Einsatz so macht.

**Reply by StevieC121176** [reply]
Vielen Dank für deine Ausführungen. Ja, momentan lasse ich mir das als Gesamtanlage anzeigen, die PV Werte stellt EVCC schon zur Verfügung.
Die Idee das über die MPPT Trecker darzustellen finde ich gut.

Aktuell habe ich aber massiv Probleme seit gestern Abend. EVCC bringt massiv Fehlermeldungen im Bezug auf MQTT, das die Oberfläche unbesiegbar wird. Kommentiere ich MQTT aus läuft es normal, dann bekommt Solectrus allerdings ja auch keine Daten.
Komisch ist das ich über den Tag rein garnichts geändert habe, weder an EVCC noch an Solectrus.

Ich hatte jetzt vor EVCC neu zu installieren, gemeinsam mit Solectrus auf einem Pi4B mit 8GB und den ioBroker auf einem pi4B mit 4GB. 
Mal schauen wie das läuft.

**Reply by ledermann** [reply]
Wenn die Messwerte von evcc nicht korrekt an den MQTT-Broker übermittelt werden, liegt das wohl an einem dieser beiden Komponenten. Ich selbst habe beim Testen gute Erfahrungen mit dem cloud-basierten MQTT-Broker [HiveMQ Cloud](https://www.hivemq.com/mqtt-cloud-broker/) gemacht. Dieser liegt dann zwar außerhalb der eigenen Netzwerkes und die Messwerte nehmen dadurch einen recht langen Weg, dafür muss man sich um diesen Teil der Infrastruktur aber auch nicht selbst kümmern.

SOLECTRUS (bzw. der zugehörige MQTT-Collector) empfängt die Messwerte nur per Subscribe, kommt mit evcc also gar nicht direkt in Berührung.

Bin gespannt, ob Du es wieder in Gang bekommst :)

**Reply by StevieC121176** [reply]
Für kurze Zeit ging es wieder, jetzt wieder das gleiche Problem. Nach etlichen Tests kann ich SOLECTRUS und ioBroker als Fehlerquelle eigentlich ausschließen. Daten kommen von EVCC über MQTT (Hab zum Testen den MQTT Explorer (Mac) genommen. EVCC bringt aber soviel Fehlermeldungen, das eine Bedienung der EVCC Oberfläche unmöglich ist.
Vielleicht kann hier jemand der EVCC auch nutzt mal die neueste Nightly ausprobieren.

**Reply by StevieC121176** [reply]
So geht wieder alles. Ich weiss nicht wieso, aber der ioBroker hat alle Werte von EVCC doppelt und dreifach gepublished. Hab das bereinigt und nun läuft alles.

Ich habe gesehen das Solectrus auch die Temperaturwerte der Batterie darstellen kann. Ich habe einen BYD Adapter in ioBroker der diese Werte liefert. Jetzt muss ich mich nur noch schlau machen wie ich die Werte über MQTT zugänglich mache.

**Reply by ledermann** [reply]
Seltsame Sache, mal abwarten, wie sich das langfristig entwickelt. 

Zu den Temperaturwerten: Wenn Du die Temperatur des Speicherns in ein MQTT-Topic gebracht hast, kannst Du dieses vom MQTT-Collector abholen lassen, sodass sie in SOLECTRUS dargestellt werden.

Die relevante Zeile in der `.env` ist die hier:

```
MQTT_TOPIC_CASE_TEMP=hier/das/topic/angeben
```

Alle von SOLECTRUS unterstützten Werte, für die man also jeweils ein Topic konfigurieren kann, findet sich hier:
https://github.com/solectrus/mqtt-collector/blob/develop/.env.example

**Reply by StevieC121176** [reply]
Hat funktioniert, Temperaturwerte werden angezeigt.
Ich habe noch 2 weitere Werte gefunden:
```
MQTT_TOPIC_BAT_CHARGE_CURRENT
MQTT_TOPIC_BAT_VOLTAGE
```
wo werden diese angezeigt?

**Reply by ledermann** [reply]
Top! 

Die beiden Batterie-Werte (Batterie-Lade-Strom in A und Batterie-Lade-Spannung in V) werden derzeitig von SOLECTRUS nirgendwo angezeigt. Höchstwahrscheinlich fliegen sie irgendwann auch ganz raus, weil sie dem normalen User kaum einen Mehrwert bringen.

**Comment by Zeppelin-PV** [comment]
der Batterie Ladestrom wäre u.U. schon interesannt;
bei Senec muss man damit rechnen dass sie immer wiedermal am Lademanagement herumschrauben um ihre Akkus bestmöglich zu schonen
um einen "thermal runaway" zu verhindern.
Ich betrachte die Ladevorgänge ständig mit einem Argusauge nachdem meine Anlage vom 19.3. bis 31.5. auf 70% lief
Im Portal "mein-Senec.de" kann man sich die Akkuwerte anschauen.
![grafik](https://github.com/solectrus/solectrus/assets/126392805/83b8c154-956f-43c1-8ec6-df609659a8f3)

**Comment by Zeppelin-PV** [comment]
![grafik](https://github.com/solectrus/solectrus/assets/126392805/2aecb228-8b38-4035-b37b-8ef4a94a0451)

... vielleicht liesen sich die zwei Akku Werte in das Anzeigefeld implementieren
Akkuladung AC seitig
Akku Status Volt
Akku Status Strom

**Reply by StevieC121176** [reply]
man könnte sie auch in die Tooltip Anzeige packen.


![IMG_0787](https://github.com/solectrus/solectrus/assets/113791299/83b1d841-d742-4f15-83d3-e38a66a4d274)

**Comment by ledermann** [comment]
Danke für die kreativen Vorschläge, ich schätze das sehr! 

In SOLECTRUS möchte ich den Schwerpunkt jedoch auf leicht verständliche Angaben wie Energie (kWh) und Kosten (€) legen. Stromstärke und -spannung sind aus meiner Sicht technische Details, die ich außen vor lassen möchte. Und ich möchte SOLECTRUS auch nicht mit Informationen überfrachten, sondern bewusst einfach halten. Ein User hatte mir mal geschrieben, dass SOLECTRUS einen hohen [WAF](https://de.wikipedia.org/wiki/Woman_acceptance_factor) habe 😄 Den möchte ich gerne beibehalten.

**Reply by StevieC121176** [reply]
:-) Der WAF ist hoch, das kann ich bestätigen. Bei dem ganzen Kostal Cloud Gedöns hat sich meine Frau nicht einmal für unsere Solaranlage interessiert. Jetzt schaut sie sogar schon selbst nach den Verbrauchs und Erzeugungswerten. Da bin ich echt begeistert.
